tools/release-plugin.sh

```diff
@@ -12,128 +13,437 @@ BASE=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+	yellow "Release script complete!"
+
+	echo ''
+	echo 'Next you need to merge the above PR into trunk.'
+
+	AUTO=()
+	MANUALTAG=()
+	MANUALTAGONLY=()
+	MANUALPUB=()
+	MANUALBOTH=()
+	for PLUGIN in "${!PROJECTS[@]}"; do
+		F="$BASE/projects/$PLUGIN/composer.json"
+		if ! jq -e '.extra["mirror-repo"] // false' "$F" &>/dev/null; then
+			continue
+		fi
+
+		if ! jq -e '.extra["wp-plugin-slug"] // .extra["wp-theme-slug"] // false' "$F" &>/dev/null; then
+			if ! jq -e '.extra["autotagger"]' "$F" &>/dev/null; then
+				MANUALTAGONLY+=( "$PLUGIN" )
+			fi
+			continue
+		fi
+
+		if jq -e '.extra["autotagger"]' "$F" &>/dev/null; then
+			if jq -e '.extra["wp-svn-autopublish"] // false' "$F" &>/dev/null; then
+				AUTO+=( "$PLUGIN" )
+			else
+				MANUALPUB+=( "$PLUGIN" )
+			fi
+		else
+			if jq -e '.extra["wp-svn-autopublish"] // false' "$F" &>/dev/null; then
+				MANUALTAG+=( "$PLUGIN" )
+			else
+				MANUALBOTH+=( "$PLUGIN" )
+			fi
+		fi
+	done
+
+	if [[ ${#AUTO[@]} -gt 0 ]]; then
+		cat <<-EOM
+
+		For these plugins: ${AUTO[*]}
+		The release will shortly be tagged to GitHub and released to SVN and you can
+		then smoke test the release. Once ready, use \`./tools/stable-tag.sh <plugin>\`
+		to update the stable tag, and you're done!
+		EOM
+	fi
+
+	if [[ ${#MANUALTAGONLY[@]} -gt 0 ]]; then
+		cat <<-EOM
+
+		For these plugins: ${MANUALTAGONLY[*]}
+		Wait for the changes to appear in the mirror repo and conduct a GitHub
+		release. Then you're done!
+		EOM
+	fi
+
+	if [[ ${#MANUALTAG[@]} -gt 0 ]]; then
+		cat <<-EOM
+
+		For these plugins: ${MANUALTAG[*]}
+		Wait for the changes to appear in the mirror repo and conduct a GitHub
+		release. The changes will then be automatically released to SVN and you can
+		then smote test the release. Once ready, use \`./tools/stable-tag.sh <plugin>\`
+		to update the stable tag, and you're done!
+		EOM
+	fi
+
+	if [[ ${#MANUALPUB[@]} -gt 0 ]]; then
+		cat <<-EOM
+
+		For these plugins: ${MANUALPUB[*]}
+		The release will shortly be tagged to GitHub. Once the tag appears, deploy it
+		to SVN by running \`./tools/deploy-to-svn.sh <plugin> <tag>\`, and smoke test.
+		When ready, flip the stable tag with \`./tools/stable-tag.sh <plugin>\` and
+		you're all set.
+		EOM
+	fi
+
+	if [[ ${#MANUALBOTH[@]} -gt 0 ]]; then
+		cat <<-EOM
+
+		For these plugins: ${MANUALBOTH[*]}
+		Wait for the changes to appear in the mirror repo, and conduct a GitHub
+		release. Next, deploy the tag to SVN by running
+		\`./tools/deploy-to-svn.sh <plugin> <tag>\`, and smoke test. When ready, flip
+		the stable tag with \`./tools/stable-tag.sh <plugin>\` and you're all set.
+		EOM
+	fi
+}
+
+preflight_checks
+
+# No args, help flag, or invalid flag, so show usage.
+if [[ $# -eq 0 || $1 == '-h' || $1 == '--help' ]]; then
+	usage
+elif [[ $1 == '-s' || $1 == '--step' ]]; then
+	if [[ $2 =~ ^[0-9]$ ]]; then
+		CUR_STEP="$2"
+		shift 2
 	else
-		PROJECTS["$SLUG"]="${PROJECTS[$PLUGIN]}"
-		unset "PROJECTS[$PLUGIN]"
+		usage
 	fi
-done
+fi
```

Comment:
> instead of having to be the only option.

More accurately: instead of having to be the first option.

Ideally BSD `getopt` would support long options. Gotta love the usage info on macOS:
```
$ getopt -h 
 --
```

The example you've listed gets a bit more complicated due to the version being optional, as the args could be `jetpack --help`, which would try to detect a version. Obviously a workaround would be to check if $1 or $2 was `--help`, which isn't pretty but is doable.

But then you have `jetpack -s 8` vs. `jetpack 19.2 -s 8`. If `$2` is `-s` then `$3` is the step number, and if `$1` is `-s` then `$2` is the step number...it gets rather ugly fast.

The version currently in `trunk` doesn't handle it unless the first option either; `getopts` fails as soon as a positional param is found, so the behaviour is the same as always.

At this stage I'd rather just keep positional args after flags unless you feel strongly about it.